### ISSUE TITLE
Error while reading h5ad file

### ISSUE THREAD

**Issue by onahman** [almaan/sepal#1]
Hi,
I got **Visium** data. I read it with scanpy and save it to h5ad format with anndata.write()
When I attempted to load the file to sepal, I got an error for not having the field 'name' (KeyError: 'name') in here:
_datasets.py line 188_
```
            self._cnt = pd.DataFrame(tmp.X,
                                    index = tmp.obs.index,
                                    columns = tmp.var['name'].values,
                                    ) 
```
It required 2 changes for it to run without errors:
```
            self._cnt = pd.DataFrame.sparse.from_spmatrix(tmp.X,
                                    index = tmp.obs.index,
                                    columns = tmp.var_names.values,
                                    ) 
```
* This might happen because of different versions of packages. I didn't check.

**Comment by almaan** [comment]
Hi @onahman ,

Thank you for reporting this issue. If you check the [supported file formats](https://github.com/almaan/sepal#supported-files-and-format) `sepal` is currently not built with support for for the scanpy anndata format but rather the one described [here](https://github.com/almaan/space2h5ad/). Have you tried using this and encountered similar issues as those you report above?

Best
Alma

**Comment by onahman** [comment]
Hi,
Thank you for the link!
I will try to use the format there.
Also in relation to my other question, both (x,y) and (_x,_y) are a requirement?

**Comment by almaan** [comment]
I'm happy to help!

You should only need the x and y coordinates, however if you have `spaceranger` output files and simply run the `space2h5ad` CLI application with that folder as input a `.h5ad` file formatted exactly as required by `sepal` will be rendered in no time at all, sparing you from any modification of you own. Of course, if you want this to be a part of a larger analysis workflow I can see how modifications might be the easiest way to go.

I'm planning on: (i) making sepal compatible with scanpy visium format and (ii) make sepal available as a module in squidpy but none of these items were available at the time of submission, hence I've waited with their implementations until publication of the manuscript.

Regarding your other issue, you should not have to have both array and pixel coordiantes and it was indeed a small bug that you spotted (thank you!), I'm taking care of it as we speak (write). I'll also see if I can't push a quick fix to the scanpy support.

Thank you!

**Comment by onahman** [comment]
You were spot on. I was hoping to use this as part of a bigger pipeline.
The field is indeed evolving quickly and the amount of formatting and tools
is hard to follow.
If you can make it compatible with scanpy that would be great! I hope that
my comments would assist your effort.
Anyway, I am glad I could help.

**Comment by almaan** [comment]
For sure! It's also about standards, up until recently there hasn't been a good "recommended" format for the Visium data - but with the introduction of the `squidpy` suite in conjunction with `scanpy`' snow more established format this has changed. In a month or so `sepal` should be compatible with the aforementioned suite.

I've pushed some changes that should - hopefully - support loading files saved in a `scanpy` h5ad format. Feel free to try it out, and I'd appreciate any feedback on whether it works or not!

**Comment by onahman** [comment]
Works perfectly!
Thank you very much :)

**Comment by almaan** [comment]
Thank you for the feedback, great to know that it's working! I will take the liberty to close this issue now.

Good luck with the analysis!
